src/app/services/User.service.ts

```diff
@@ -28,15 +26,13 @@ export class UserService {
     return this.userMap.toDTO(user);
   }
   async create(userDTO: IUserAPIincomingDTO) {
-    const id = v4();
-    const userToSave: IUserPersistenceDTO = {
-      id,
+    const userToSave: User = new User({
```

Comment:
No need for a type declaration here, ts will infer the proper type from new User